src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -939,11 +939,20 @@ void Connect(NpgsqlTimeout timeout)
 
     async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationToken)
     {
-        // Note that there aren't any timeout-able or cancellable DNS methods
+        Task<IPAddress[]> GetHostAddressesAsync(CancellationToken ct) =>
+#if NET6_0_OR_GREATER
+            Dns.GetHostAddressesAsync(Host, ct);
+#else
+            Dns.GetHostAddressesAsync(Host);
+#endif
+
+        // Whether the framework and/or the OS platform support Dns.GetHostAddressesAsync cancellation API or they do not,
+        // we always fake-cancel the operation with the help of TaskTimeoutAndCancellation.WaitAsync. It stops waiting
+        // and raises the exception, while the actual task may be left running.
         var endpoints = NpgsqlConnectionStringBuilder.IsUnixSocket(Host, Port, out var socketPath)
-            ? new EndPoint[] { new UnixDomainSocketEndPoint(socketPath) }
-            : (await GetHostAddressesAsync(timeout, cancellationToken))
-            .Select(a => new IPEndPoint(a, Port)).ToArray();
+                ? new EndPoint[] { new UnixDomainSocketEndPoint(socketPath) }
```

Comment:
indentation

Reply:
This is how my IDE did it. Do you want me to add more idents or remove any?

```
        var endpoints = NpgsqlConnectionStringBuilder.IsUnixSocket(Host, Port, out var socketPath)
                                ? new EndPoint[] { new UnixDomainSocketEndPoint(socketPath) }
                                : Dns.GetHostAddresses(Host).Select(a => new IPEndPoint(a, Port)).ToArray();
```

v.s.
```
        var endpoints = NpgsqlConnectionStringBuilder.IsUnixSocket(Host, Port, out var socketPath)
        ? new EndPoint[] { new UnixDomainSocketEndPoint(socketPath) }
        : Dns.GetHostAddresses(Host).Select(a => new IPEndPoint(a, Port)).ToArray();
```

Reply:
We simply do 4-space increases, so:

```c#
        var endpoints = NpgsqlConnectionStringBuilder.IsUnixSocket(Host, Port, out var socketPath)
            ? new EndPoint[] { new UnixDomainSocketEndPoint(socketPath) }
            : (await TaskTimeoutAndCancellation.WaitAsync(GetHostAddressesAsync, timeout, cancellationToken))
            .Select(a => new IPEndPoint(a, Port)).ToArray();
```

Reply:
Got it.